File: 1448. Count Good Nodes in Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function(root) {
    let result = 0;
    const stack = [[root, root.val]];
    while(stack.length) {
        let [node, maxAncestor] = stack.shift();
        if (node) {
            if (node.val >= maxAncestor) {
                result++;
                maxAncestor = node.val;
            }
            if (node.left) {
                stack.push([node.left, maxAncestor])
            }
            if (node.right) {
                stack.push([node.right, maxAncestor])
            }
        }
    }
    return result;
};

File: Golang/Tree/1448.Count-good-nodes-in-binary-tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func goodNodes(root *TreeNode) int {
	ret := 0
	var helper func(node *TreeNode, max int)
	helper = func(node *TreeNode, max int) {
		if node != nil {
			if node.Val >= max {
				ret++
				max = node.Val
			}
			helper(node.Left, max)
			helper(node.Right, max)
		}
	}
	helper(root, root.Val)
	return ret
}
